### PR TITLE
DEV: reduces jquery usage and memory leaks in composer

### PR DESCRIPTION
**Pull request by jjaffeux**
Removes more than 60 jquery function leaks in one `Acceptance: Composer` run.

### CODE REVIEW

**Comment by jjaffeux** [comment]
I guess these could probably be refactor even further some day, but this is probably a good first step.

File: app/assets/javascripts/discourse/app/components/composer-body.js [hunk 1]
```diff
@@ -207,11 +233,18 @@ export default Component.extend(KeyEnterEscape, {
...
     this._super(...arguments);
+
     this.appEvents.off("composer:resize", this, this.resize);
     if (this._visualViewportResizing()) {
       window.visualViewport.removeEventListener("resize", this.viewportResize);
     }
 
+    START_DRAG_EVENTS.forEach((startDragEvent) => {
+      this.element
+        ?.querySelector(".grippie")
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/composer-body.js:244]: Minor: are there any cases where `this.element` is falsy in `willDestroyElement()`?

**Review by CvX — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/composer-body.js (hunk 1 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/components/composer-body.js:244]: Yeh...... I tend to be over paranoid this we have ?.. I guess it's not needed indeed

File: app/assets/javascripts/discourse/app/components/composer-editor.js [hunk 2]
```diff
@@ -138,9 +139,7 @@ export default Component.extend(ComposerUpload, {
 
   @discourseComputed
   showLink() {
-    return (
-      this.currentUser && this.currentUser.get("link_posting_access") !== "none"
-    );
+    return this.currentUser?.link_posting_access !== "none";
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/composer-editor.js:142]: I think this is an incorrect use of the optional chaining operator:

```js
let dis = { currentUser: { link_posting_access: "none" } };
console.log(dis.currentUser && dis.currentUser.link_posting_access !== "none");
console.log(dis.currentUser?.link_posting_access !== "none");
// false and false, all good

dis = { currentUser: null };
console.log(dis.currentUser && dis.currentUser.link_posting_access !== "none");
console.log(dis.currentUser?.link_posting_access !== "none");
// null and true, oops
```

**Review by CvX — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/discourse/app/components/composer-editor.js [hunk 3]
```diff
@@ -218,34 +218,42 @@ export default Component.extend(ComposerUpload, {
...
+      const input = this.element.querySelector(".d-editor-input");
+      const preview = this.element.querySelector(".d-editor-preview-wrapper");
+      this._initInputPreviewSync(input, preview);
     } else {
-      $input.on("scroll", () =>
-        throttle(this, this._syncEditorAndPreviewScroll, $input, $preview, 20)
-      );
+      this.element
+        .querySelector(".d-editor-input")
+        ?.addEventListener("scroll", this._throttledSyncEditorAndPreviewScroll);
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/composer-editor.js:256]: Should this event listener be removed at some point? (e.g. `willDestroyElement`)

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/components/composer-editor.js:256]: Yes indeed, that should do it: https://github.com/discourse/discourse/pull/14924/commits/d016cd2b340c706cb662e2d9767bebecccf067cc

**Review by CvX — APPROVED** [review]

**Pull request merged**